test/analysis/test_curve_fit.py

```diff
@@ -0,0 +1,212 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test curve fitting base class."""
+
+from typing import List, Callable
+
+import numpy as np
+from qiskit.test import QiskitTestCase
+
+from qiskit_experiments import ExperimentData
+from qiskit_experiments.analysis import CurveAnalysis, SeriesDef
+from qiskit_experiments.base_experiment import BaseExperiment
+
+
+class FakeExperiment(BaseExperiment):
+    """A fake experiment class."""
+
+    def __init__(self):
+        super().__init__(qubits=(0,), experiment_type="fake_experiment")
+
+    def circuits(self, backend=None, **circuit_options):
+        return []
```

Comment:
We should have this in a separate file to reuse it.

Reply:
Okey, I'll open separate PR :)

Reply:
See #49